src/Build/Evaluation/Evaluator.cs

```diff
@@ -624,7 +624,7 @@ private void Evaluate()
                 }
 
                 _data.InitialTargets = initialTargets;
-                MSBuildEventSource.Log.EvaluatePass1Stop(projectFile);
```

Comment:
I don't think this is accurate because _projectRootElement.Properties does not include properties of imported projects.

Also, since it does a nontrivial amount of work, please surround it with an IsEnabled check.

Reply:
The `ProjectImportGroupElement` is a `ProjectElementContainer` and that should get included in both the `Properties `recursion as it is with the `PerformDepthFirstPass `recursion. Also the `Count `is a property get on an already instantiated  `ReadOnlyCollection` (as a part of the actual evaluation above the log). No?

Reply:
We talked about this offline, but ProjectImportElement is not a ProjectElementContainer so isn't included, and accessing the Properties ReadOnlyCollection constructs it as needed, hence the perf hit.